internal/gengapic/genrest.go

```diff
@@ -309,15 +314,18 @@ func (g *generator) generateQueryString(m *descriptor.MethodDescriptorProto) {
 	requestObject := "req"
 
 	p("params := url.Values{}")
-	// TODO(dovs): make this work for nested fields
 	for _, path := range fields {
 		field := queryParams[path]
 		accessor := buildAccessor(path)
 		closer := "}"
 		if field.GetLabel() == descriptor.FieldDescriptorProto_LABEL_REPEATED {
 			// It's a slice, so check for nil
 			p("if %s%s != nil {", requestObject, accessor)
-		} else if field.GetLabel() == descriptor.FieldDescriptorProto_LABEL_OPTIONAL {
+		} else if field.GetLabel() == descriptor.FieldDescriptorProto_LABEL_REQUIRED {
```

Comment:
Ah so `google.api.field_behavior` is an annotation. The label in proto3 is different. We shouldn't be using this one. If you want to see if its annotated as required, check for the annotation. But if you want to check for `proto3_optional`, use the `GetProto3Optional()` API.

Reply:
Damn, I always forget this